Lib/asyncio/sslproto.py

```diff
@@ -590,12 +590,7 @@ def _on_handshake_complete(self, handshake_exc):
                 raise handshake_exc
 
             peercert = sslobj.getpeercert()
-            if not hasattr(self._sslcontext, 'check_hostname'):
-                # Verify hostname if requested, Python 3.4+ uses check_hostname
-                # and checks the hostname in do_handshake()
-                if (self._server_hostname and
-                        self._sslcontext.verify_mode != ssl.CERT_NONE):
-                    ssl.match_hostname(peercert, self._server_hostname)
+            # Since 3.7, the hostname is matched by OpenSSL during handshake.
```

Comment:
Not sure this comment is needed here, since it's not acually referring to any code which exists